### PR TITLE
Add sphinx documentation of the I/O protocols available in Sidre

### PR DESCRIPTION
**Pull request by nselliott**
# Summary

- This PR is a documentation
- It does the following (modify list as needed):
  - Adds documentation of the I/O protocols used in Sidre
  - Addresses one of the checkboxes in #1019

### CODE REVIEW

File: src/axom/sidre/docs/sphinx/file_io.rst [hunk 1]
```diff
@@ -59,3 +59,65 @@ Please see `Sidre API Documentation <../../../../doxygen/html/sidretop.html>`_
...
+    scalar and string Views. When loading scalars or strings, the library will
+    determine implicitly what type to use, which may not be the same type
+    used in the original objects that were saved. For example, the value of
+    a 32-bit integer will be saved in plain text with no information about
+    its original bit width. When loading, the library may read that value
+    into a 64-bit integer.
+
+``conduit_hdf5``
+
+    This saves group as a conduit node hierarchy. Datatypes are preserved, but
```
Comment (kennyweiss) [inline src/axom/sidre/docs/sphinx/file_io.rst:99]: Possible typo:

```suggestion
    This saves the group as a conduit node hierarchy. Datatypes are preserved, but
```

File: src/axom/sidre/docs/sphinx/file_io.rst [hunk 2]
```diff
@@ -59,3 +59,65 @@ Please see `Sidre API Documentation <../../../../doxygen/html/sidretop.html>`_
...
+``conduit_json``
+
+    This is the same as ``conduit_hdf5`` except it uses JSON text. The same
+    caveats about floating-point accuracy apply here as in the other JSON
+    protocols.
+
+``conduit_bin``
+
+    This protocol writes data in conduit's native binary format, and also
+    produces an anccompanying JSON file that contains the schema describing
```
Comment (kennyweiss) [inline src/axom/sidre/docs/sphinx/file_io.rst:113]: Typo:
```suggestion
    produces an accompanying JSON file that contains the schema describing
```

File: src/axom/sidre/docs/sphinx/file_io.rst [hunk 3]
```diff
@@ -59,3 +59,65 @@ Please see `Sidre API Documentation <../../../../doxygen/html/sidretop.html>`_
...
+    how to parse the binary data. The hierarchal layout of the data is the
+    same as ``conduit_hdf5`` and ``conduit_json``. The binary accuracy of
+    floating-point values is preserved. 
+
+``json``
+
+    This protocol saves the smallest amount of information, as it Writes all
```
Comment (kennyweiss) [inline src/axom/sidre/docs/sphinx/file_io.rst:120]: Possible typo:

```suggestion
    This protocol saves the smallest amount of information, as it writes all
```

**Review by kennyweiss — APPROVED** [review]
Thanks @nselliott -- this is a really nice overview of the different protocols.

File: src/axom/sidre/docs/sphinx/file_io.rst [hunk 4]
```diff
@@ -59,3 +59,65 @@ Please see `Sidre API Documentation <../../../../doxygen/html/sidretop.html>`_
...
+``sidre_hdf5``
+
+    This is the protocol that supports the most features. It preserves the
+    entire Sidre hierarchy of Groups and Views, with Buffers saved separately
+    from the Views that use them. Each Buffer is saved exactly once, to avoid
+    duplicating data when a Buffer is used by more than one View. The schemas
+    describing the datatypes in each View are saved, and all numerical data is
+    saved with full binary accuracy.
+     
+    ``sidre_hdf5`` is the only protocal that supports the
```
Comment (agcapps) [inline src/axom/sidre/docs/sphinx/file_io.rst:74]:
```suggestion
    ``sidre_hdf5`` is the only protocol that supports the
```

File: src/axom/sidre/docs/sphinx/file_io.rst [hunk 5]
```diff
@@ -59,3 +59,65 @@ Please see `Sidre API Documentation <../../../../doxygen/html/sidretop.html>`_
...
+    ``loadExternalData()`` method and is the only protocoal that allows
```
Comment (agcapps) [inline src/axom/sidre/docs/sphinx/file_io.rst:75]:
```suggestion
    ``sidre_hdf5`` is the only protocol that supports the
    ``loadExternalData()`` method and is the only protocol that allows
```

**Review by agcapps — APPROVED** [review]
I appreciate the details!

**Pull request merged**